src/uag.c

```diff
@@ -1128,6 +1128,32 @@ struct ua *uag_find_requri(const char *requri)
 }
 
 
+/**
+ * Find a User-Agent (UA) best fitting for an SIP request
+ *
+ * @param requri The SIP uri pointer-length string for the request
+ *
+ * @return User-Agent (UA) if found, otherwise NULL
+ */
+struct ua *uag_find_requri_pl(const struct pl *requri)
+{
+	char *uric;
+	struct ua *ua;
+	int err;
+
+	if (!pl_isset(requri))
+		return NULL;
+
+	err = pl_strdup(&uric, requri);
```

Comment:
perhaps swap the logic so that the _pl version is the base version ?

then you can avoid strdup

Reply:
Okay, I had this already and also for `account_uri_complete()`. Your right it simplifies the code. Will revert 917038fcbe292eb73c2abafd421deba0cffb1269.